app/Listeners/Stand/RecordStandAssignmentHistory.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+namespace App\Listeners\Stand;
+
+use App\Events\StandAssignedEvent;
+use App\Models\Stand\StandAssignmentsHistory;
+use Illuminate\Support\Facades\Auth;
+
+class RecordStandAssignmentHistory
+{
+    public function handle(StandAssignedEvent $assignedEvent) : bool
+    {
+        // Mark any current assignments as deleted and create a new history item
+        StandAssignmentsHistory::where('callsign', $assignedEvent->getStandAssignment()->callsign)->delete();
+        StandAssignmentsHistory::create(
+            [
+                'callsign' => $assignedEvent->getStandAssignment()->callsign,
```

Comment:
Would there be any issues here if this method is called twice? Maybe you'd want to just assign it once and reference callsign & stand_id?

Reply:
There shouldn't be, but agreed this can be tidied - done